tensorboard/defs/vulcanize.bzl

```diff
@@ -52,17 +54,17 @@ def _tensorboard_html_binary(ctx):
                   ctx.attr.output_path,
                   ctx.outputs.html.path,
                   ignore_regexs_file_path] +
-                 [f.path for f in jslibs] +
-                 [f.path for f in manifests]),
+                 [f.path for f in jslibs.to_list()] +
+                 [f.path for f in manifests.to_list()]),
       progress_message="Vulcanizing %s" % ctx.attr.input_path)
 
   # webfiles manifest
   manifest_srcs = [struct(path=ctx.outputs.html.path,
                           longpath=long_path(ctx, ctx.outputs.html),
                           webpath=ctx.attr.output_path)]
-  manifest = ctx.new_file(ctx.configuration.bin_dir,
-                          "%s.pbtxt" % ctx.label.name)
-  ctx.file_action(
+  manifest = ctx.actions.declare_file(paths.join(
```

Comment:
@c-parsons: Is the recommended migration for #5825 really to use
`paths.join` everywhere? It’s noticeably more verbose, and the [docs for
`paths.join` indicate that it doesn’t support Windows-style paths at
all][1].

[1]: https://github.com/bazelbuild/bazel-skylib/blob/b113ed5d05ccddee3093bb157b9b02ab963c1c32/lib/paths.bzl#L17-L19

Reply:
`paths.join` shouldn't be necessary in many cases -- is there a reason you need this specific path for your declared file?
If so, you must consider using the `sibling` [parameter](https://docs.bazel.build/versions/master/skylark/lib/actions.html#declare_file) to `declare_file` instead of doing path math.

But yes, otherwise, paths.join is our best solution to generic path joining.

As for the rationale, the original API deprecation doc and decision predate my time on the team. The best original doc I could find for the migration is [here](https://docs.google.com/document/d/1wSp1ff_7-f3xXVouayo1zdv-RkzZEc31diy7lPYQTko/edit#heading=h.rl2nzsygsyps), but this was some time in 2017 :\